### ISSUE TITLE
Fresh install of Bndtools doesn't associate .bnd extension

### ISSUE THREAD

**Issue by kriegfrj** [bndtools/bnd#5797]
I haven't done any bnd development for a while and was trying to dust off my setup so that I could have a look at one of the outstanding issues. I was having a weird "circular dependency" issue with aQute.libg which I tried to resolve by upgrading my Eclipse installation to the latest. I ended up with an installation that wouldn't start.

So I decided to set up a fresh installation of Eclipse 2023-06 with the latest snapshot release of Bndtools (7.1).

When I opened and imported my first Bnd workspace, I found that the PDE plugin manifest editor had taken over the file association of my bnd.bnd file, and then failed to open it when I double-clicked on it.

![Bnd-PDE-fail](https://github.com/bndtools/bnd/assets/7224087/fc43f1b5-c162-4fd0-84bd-6aacc226fef0)

**Comment by pkriens** [comment]
holy moses ... @tjwatson @HannesWell what's this?

**Comment by kriegfrj** [comment]
The good news is it's an easy fix to re-associate the file with the Bndtools editor. :smile:

> holy moses ... @tjwatson @HannesWell what's this?
@laeubi also did something to embed bnd support in PDE recently?

**Comment by pkriens** [comment]
Yes, but I do not think we should get into a fight for this association?

**Comment by laeubi** [comment]
There is no "fight" but you probably want to make the BND editor the default:

- https://github.com/eclipse-pde/eclipse.pde/issues/672#issuecomment-1660440036

beside that Eclipse-Editor association is a mystery as you can see here:

![](https://user-images.githubusercontent.com/208716/254846348-c01fddf4-3ed3-4015-b231-752cace2ba00.png)

BND editor is even shown as having precedence over the pde editor ...

**Comment by kriegfrj** [comment]
Is there a way to make the install of the Bndtools feature so that it automatically makes the Bnd editors the defaults for their file formats (esp if no default is already set).

I just repeated the exercise and it seems to be ok this time. I'm not sure what I did different the first time: Perhaps I forgot to restart Eclipse after installing Bndtools, or maybe I accidentally made the PDE Editor the default when I opened it the first time (before Bndtools was installed).

So it seems that this may have been a false alarm. Unless there is a way to make it the default automatically when you install Bndtools, there's probably not a lot more we can do.
